core/app/core/src/lib/fields/int/templates/edit/int.component.ts

```diff
@@ -0,0 +1,87 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License
+ * version 3, these Appropriate Legal Notices must retain the display of the
+ * "Supercharged by SuiteCRM" logo. If the display of the logos is not reasonably
+ * feasible for technical reasons, the Appropriate Legal Notices must display
+ * the words "Supercharged by SuiteCRM".
+ */
+
+import { isEqual } from 'lodash-es';
+import {Component} from '@angular/core';
+import {FormControl} from '@angular/forms';
+import {BaseNumberComponent} from '../../../base/base-number.component';
+import {SystemConfigStore} from '../../../../store/system-config/system-config.store';
+import {DataTypeFormatter} from '../../../../services/formatters/data-type.formatter.service';
+import {UserPreferenceStore} from '../../../../store/user-preference/user-preference.store';
+import {FieldLogicManager} from '../../../field-logic/field-logic.manager';
+import {FieldLogicDisplayManager} from '../../../field-logic-display/field-logic-display.manager';
+
+@Component({
+    selector: 'scrm-int-edit',
+    templateUrl: './int.component.html',
+    styleUrls: []
+})
+export class IntEditFieldComponent extends BaseNumberComponent {
+    public intValue = 0;
+    public intValueFormControl: FormControl = new FormControl(this.intValue);
+
+    constructor(
+        protected userPreferences: UserPreferenceStore,
+        protected systemConfig: SystemConfigStore,
+        protected typeFormatter: DataTypeFormatter,
+        protected logic: FieldLogicManager,
+        protected logicDisplay: FieldLogicDisplayManager,
+    ) {
+        super(userPreferences, systemConfig, typeFormatter, logic, logicDisplay);
+    }
+
+    ngOnInit(): void {
+        this.subscribeIntValueChanges();
+
+        super.ngOnInit();
+        this.subscribeValueChanges();
+    }
+
+    ngOnDestroy(): void {
+        this.unsubscribeAll();
+    }
+
+    private subscribeIntValueChanges(): void {
+        this.subs.push(
+            this.intValueFormControl.valueChanges.subscribe(valueChanges => {
+                this.intValue = valueChanges;
+                if (!this.field.formControl || isEqual(this.field.formControl.value, valueChanges)) {
+                    return;
+                }
+                this.field.formControl.setValue(valueChanges.toString());
+            })
+        );
+        this.subs.push(
+            this.field.valueChanges$.subscribe((fieldValue) => {
+                const intValue = parseInt(fieldValue.value,10) || 0;
+                if (isEqual(this.intValueFormControl.value, intValue)) {
+                    return;
+                }
+                this.intValueFormControl.setValue(intValue);
+            })
+        );
+    }
```

Comment:
Hi @mpuyosa91 couldn't we just override the base `subscribeValueChanges` and use the existing form control?